src/main/java/com/minecolonies/permissions/ColonyPermissionEventHandler.java

```diff
@@ -0,0 +1,269 @@
+        }
+    }
+
+    /**
+     * BlockEvent.BreakEvent handler.
+     *
+     * @param event BlockEvent.BreakEvent
+     */
+    @SubscribeEvent
+    public void on(final BlockEvent.BreakEvent event)
+    {
+        if (checkBlockEventDenied(event.getWorld(), event.getPos(), event.getPlayer(), event.getWorld().getBlockState(event.getPos())))
+        {
+            cancelEvent(event);
+        }
+    }
+
+    /**
+     * ExplosionEvent.Detonate handler.
+     *
+     * @param event ExplosionEvent.Detonate
+     */
+    @SubscribeEvent
+    public void on(ExplosionEvent.Detonate event)
+    {
+        Log.getLogger().info("ExplosionEvent.Detonate");
+
+        // if block is in colony -> remove from list
+        for (BlockPos pos : event.getAffectedBlocks())
+        {
+            if (colony.isCoordInColony(event.getWorld(), pos))
+            {
+                Log.getLogger().info("Found affected block in colony, removing from affected list");
+                event.getAffectedBlocks().remove(pos);
+            }
+        }
+
+        // if entity is in colony -> remove from list
+        for (Entity entity : event.getAffectedEntities())
+        {
+            if (colony.isCoordInColony(entity.getEntityWorld(), entity.getPosition()))
+            {
+                Log.getLogger().info("Found affected entity in colony, removing from affected list");
+                event.getAffectedEntities().remove(entity);
+            }
+        }
+    }
+
+    /**
+     * ExplosionEvent.Start handler.
+     *
+     * @param event ExplosionEvent.Detonate
+     */
+    @SubscribeEvent
+    public void on(ExplosionEvent.Start event)
+    {
+        Log.getLogger().info("ExplosionEvent.Start");
+        if (colony.isCoordInColony(event.getWorld(), new BlockPos(event.getExplosion().getPosition())))
+        {
+            Log.getLogger().info("Explosion in colony, removing!");
+            cancelEvent(event);
+        }
+
+    }
+
+    /**
+     * PlayerInteractEvent handler.
+     * <p>
+     * Check, if a player right clicked a block.
+     * Deny if:
+     * - If the block is in colony
+     * - block is AbstractBlockHut
+     * - player has not permission
+     *
+     * @param event PlayerInteractEvent
+     */
+    @SubscribeEvent
+    public void on(final PlayerInteractEvent event)
+    {
+        Log.getLogger().info("Check if coordinate is in colony `" + this.colony.getName() + "`");
+        if (colony.isCoordInColony(event.getWorld(), event.getPos()))
+        {
+            Log.getLogger().info("Rightclick Coordinate is in `" + this.colony.getName() + "`");
+
+            final Block block = event.getWorld().getBlockState(event.getPos()).getBlock();
+
+            // Huts
+            if (block instanceof AbstractBlockHut &&
+                  !colony.getPermissions().hasPermission(event.getEntityPlayer(), Permissions.Action.ACCESS_HUTS))
+            {
+                cancelEvent(event);
+            }
+        }
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)